diff -uNr wxmozilla-0.5.6/configure.in wxmozilla-0.5.6.new/configure.in
--- wxmozilla-0.5.6/configure.in	2006-04-23 16:33:16.000000000 +0200
+++ wxmozilla-0.5.6.new/configure.in	2006-12-10 22:53:50.000000000 +0100
@@ -1,7 +1,7 @@
 # Process this file with autoconf to produce a configure script
 AC_PREREQ([2.52])
 
-AC_INIT([wxMozilla], [0.5.6], [wxmozilla-devel@lists.sourceforge.net])
+AC_INIT([wxMozilla], [0.5.7], [wxmozilla-devel@lists.sourceforge.net])
 AC_CONFIG_SRCDIR([src/wxMozillaBrowser.cpp])
 
 AC_CANONICAL_BUILD
@@ -12,6 +12,7 @@
 
 build_python=0
 use_firefox=0
+use_seamonkey=0
 
 AC_ARG_ENABLE([python],
               AS_HELP_STRING([--enable-python],
@@ -21,6 +22,10 @@
               AS_HELP_STRING([--enable-firefox],
                              [build against Firefox instead of Mozilla]),
               [use_firefox="$enableval"])
+AC_ARG_ENABLE([seamonkey],
+              AS_HELP_STRING([--enable-seamonkey],
+                             [build against Seamonkey instead of Mozilla (1.7-)]),
+              [use_seamonkey="$enableval"])
 
 AM_INIT_AUTOMAKE
 AM_MAINTAINER_MODE
@@ -90,6 +95,11 @@
 	[  --with-mozilla-home=dir Where mozilla libs are at runtime ],
 	[ MOZILLA_HOME="$withval" ])
 
+dnl "0" means "not used"
+WXMOZ_FIREFOX_VERSION=0
+WXMOZ_SEAMONKEY_VERSION=0
+WXMOZ_MOZILLA_VERSION=0
+
 if test "x$mozilla_includes" != "x" ; then
     dnl Don't detect mozilla, use given version
     if test "x$mozilla_libs" = "x" ; then
@@ -124,6 +134,26 @@
             MOZILLA_CFLAGS="$MOZILLA_CFLAGS $MOZILLA_GTK_CFLAGS"
             MOZILLA_LIBS="$MOZILLA_LIBS $MOZILLA_GTK_LIBS"
         fi
+
+        AC_MSG_CHECKING([for Firefox version])
+        WXMOZ_FIREFOX_VERSION="`$PKG_CONFIG --modversion firefox-gtkmozembed | cut -d. -f1-2`"
+        AC_MSG_RESULT([$WXMOZ_FIREFOX_VERSION])
+        WXMOZ_FIREFOX_VERSION="`echo $WXMOZ_FIREFOX_VERSION | sed -e 's/\.//g'`"
+    elif test "$use_seamonkey" = "yes" ; then
+        PKG_CHECK_MODULES([MOZILLA], 
+            [mozilla-nspr >= 1.0 seamonkey-xpcom])
+        mozilla_includes=`$PKG_CONFIG --variable=includedir seamonkey-xpcom`
+        mozilla_libs=`$PKG_CONFIG --variable=libdir seamonkey-xpcom`
+        if test $USE_WXGTK = "yes"; then
+            PKG_CHECK_MODULES([MOZILLA_GTK], [seamonkey-gtkmozembed >= 1.0])
+            MOZILLA_CFLAGS="$MOZILLA_CFLAGS $MOZILLA_GTK_CFLAGS"
+            MOZILLA_LIBS="$MOZILLA_LIBS $MOZILLA_GTK_LIBS"
+        fi
+
+        AC_MSG_CHECKING([for Seamonkey version])
+        WXMOZ_SEAMONKEY_VERSION="`$PKG_CONFIG --modversion seamonkey-gtkmozembed | cut -d. -f1-2`"
+        AC_MSG_RESULT([$WXMOZ_SEAMONKEY_VERSION])
+        WXMOZ_SEAMONKEY_VERSION="`echo $WXMOZ_SEAMONKEY_VERSION | sed -e 's/\.//g'`"
     else
         PKG_CHECK_MODULES([MOZILLA], 
             [mozilla-nspr >= 1.3 mozilla-xpcom])
@@ -134,8 +164,17 @@
             MOZILLA_CFLAGS="$MOZILLA_CFLAGS $MOZILLA_GTK_CFLAGS"
             MOZILLA_LIBS="$MOZILLA_LIBS $MOZILLA_GTK_LIBS"
         fi
+        AC_MSG_CHECKING([for Mozilla version])
+        WXMOZ_MOZILLA_VERSION="`$PKG_CONFIG --modversion mozilla-gtkmozembed | cut -d. -f1-2`"
+        AC_MSG_RESULT([$WXMOZ_MOZILLA_VERSION])
+        WXMOZ_MOZILLA_VERSION="`echo $WXMOZ_MOZILLA_VERSION | sed -e 's/\.//g'`"
     fi
 fi
+
+MOZILLA_CFLAGS="$MOZILLA_CFLAGS -DWXMOZ_FIREFOX_VERSION=$WXMOZ_FIREFOX_VERSION"
+MOZILLA_CFLAGS="$MOZILLA_CFLAGS -DWXMOZ_SEAMONKEY_VERSION=$WXMOZ_SEAMONKEY_VERSION"
+MOZILLA_CFLAGS="$MOZILLA_CFLAGS -DWXMOZ_MOZILLA_VERSION=$WXMOZ_MOZILLA_VERSION"
+
 dnl Additional mozilla settings:
 if test "x$MOZILLA_HOME" = "x"; then
     MOZILLA_HOME=$mozilla_libs
@@ -144,7 +183,7 @@
 for subdir in unix dom windowwatcher webbrowserpersist appcomps find txtsvc \
               docshell shistory webshell gfx necko widget content layout \
               browser webbrwsr uconv embed_base uriloader editor \
-              commandhandler composer pref profdirserviceprovider ; do
+              commandhandler composer pref profile profdirserviceprovider ; do
     MOZILLA_CFLAGS="$MOZILLA_CFLAGS -I$mozilla_includes/$subdir"
 done
 if test $USE_WXGTK = "yes"; then
diff -uNr wxmozilla-0.5.6/src/helpers.cpp wxmozilla-0.5.6.new/src/helpers.cpp
--- wxmozilla-0.5.6/src/helpers.cpp	2004-08-15 11:31:12.000000000 +0200
+++ wxmozilla-0.5.6.new/src/helpers.cpp	2006-12-10 22:53:50.000000000 +0100
@@ -115,10 +115,10 @@
 
 nsIURI* wxString_to_nsURI(const wxString from)
 {
-  nsIURI* url;  
   nsCString utf8_buffer = NS_ConvertUCS2toUTF8(wxString_to_nsString(from, wxConvLocal));//@todo YORIC : is this the right converter ?
   
-  nsresult  rv                   = nsComponentManager::CreateInstance(NS_SIMPLEURI_CONTRACTID, nsnull, NS_GET_IID(nsIURI), (void**)&url);
+	nsresult rv;
+	nsCOMPtr<nsIURI> url = do_CreateInstance( NS_SIMPLEURI_CONTRACTID, &rv );
   wxASSERT_MSG(NS_SUCCEEDED(rv), wxT("Could not create an instance of the url"));
   rv = url->SetSpec(nsCAutoString(utf8_buffer));
   wxASSERT_MSG(NS_SUCCEEDED(rv), wxT("Could not create set specs for the url"));
diff -uNr wxmozilla-0.5.6/src/Makefile.am wxmozilla-0.5.6.new/src/Makefile.am
--- wxmozilla-0.5.6/src/Makefile.am	2005-05-17 21:05:18.000000000 +0200
+++ wxmozilla-0.5.6.new/src/Makefile.am	2006-12-10 22:53:50.000000000 +0100
@@ -1,5 +1,6 @@
 
-AM_CXXFLAGS = $(MOZILLA_CFLAGS) $(WX_CXXFLAGS) $(GTK_CFLAGS) -fno-rtti
+AM_CXXFLAGS = $(MOZILLA_CFLAGS) $(WX_CXXFLAGS) $(GTK_CFLAGS) -fno-rtti \
+			  -DMOZILLA_INTERNAL_API
 
 lib_LTLIBRARIES = lib@WXMOZILLALIB@.la
 
diff -uNr wxmozilla-0.5.6/src/wxMozillaBrowserChrome.cpp wxmozilla-0.5.6.new/src/wxMozillaBrowserChrome.cpp
--- wxmozilla-0.5.6/src/wxMozillaBrowserChrome.cpp	2005-06-02 14:50:56.000000000 +0200
+++ wxmozilla-0.5.6.new/src/wxMozillaBrowserChrome.cpp	2006-12-10 22:53:50.000000000 +0100
@@ -40,17 +40,17 @@
     {
         case 0:
             thisEvent.m_leftDown   = true;
-            thisEvent.m_eventType  = eventTypes[0];
+            thisEvent.SetEventType(eventTypes[0]);
             break;
 
         case 1:
             thisEvent.m_middleDown = true;
-            thisEvent.m_eventType  = eventTypes[1];
+            thisEvent.SetEventType(eventTypes[1]);
             break;
 
         default:
             thisEvent.m_rightDown  = true;
-            thisEvent.m_eventType  = eventTypes[2];
+            thisEvent.SetEventType(eventTypes[2]);
     }
 
     PRBool result;
@@ -1485,7 +1485,7 @@
         m_mouseInWindow = true;
 
         wxMouseEvent thatEvent(thisEvent);
-        thatEvent.m_eventType = wxEVT_ENTER_WINDOW;
+        thatEvent.SetEventType(wxEVT_ENTER_WINDOW);
         mMoz->GetEventHandler()->ProcessEvent(thatEvent);
     }
 
diff -uNr wxmozilla-0.5.6/src/wxMozillaBrowserChrome.h wxmozilla-0.5.6.new/src/wxMozillaBrowserChrome.h
--- wxmozilla-0.5.6/src/wxMozillaBrowserChrome.h	2005-05-04 09:42:58.000000000 +0200
+++ wxmozilla-0.5.6.new/src/wxMozillaBrowserChrome.h	2006-12-10 22:53:50.000000000 +0100
@@ -41,6 +41,7 @@
 #include "nsIDOMMouseListener.h"
 #include "nsIDOMKeyListener.h"
 #include "nsIURIContentListener.h"
+#include "xpcom/nsIComponentManager.h"
 
 // forward declarations
 
diff -uNr wxmozilla-0.5.6/src/wxMozillaBrowser.cpp wxmozilla-0.5.6.new/src/wxMozillaBrowser.cpp
--- wxmozilla-0.5.6/src/wxMozillaBrowser.cpp	2005-07-09 20:29:00.000000000 +0200
+++ wxmozilla-0.5.6.new/src/wxMozillaBrowser.cpp	2006-12-10 22:53:50.000000000 +0100
@@ -33,6 +33,7 @@
 #include "nsString.h"
 #include "nsCOMPtr.h"
 #include "nsIEventQueueService.h"
+#include "nsIComponentRegistrar.h"
 #include "nsIInputStream.h"
 #include "nsIAtom.h"
 #include "nsIPref.h"
@@ -294,7 +295,11 @@
         NS_ShutdownXPCOM(nsnull);
         XPCOMGlueShutdown();
 #else
+        #if WXMOZ_FIREFOX_VERSION >= 15 || WXMOZ_SEAMONKEY_VERSION >= 10
+        NS_ShutdownXPCOM(nsnull);
+        #else
 		NS_TermEmbedding();
+        #endif
 #endif
         #ifdef __WXGTK__
              gdk_input_remove(io_id);
@@ -347,7 +352,11 @@
         //nsCOMPtr<nsIServiceManager> servMan;
         NS_InitXPCOM2(nsnull, greDir, nsnull); 
 #else
+        #if WXMOZ_FIREFOX_VERSION >= 15 || WXMOZ_SEAMONKEY_VERSION >= 10
+        rv = NS_InitXPCOM3(nsnull, greDir, nsnull, nsnull, nsnull);
+        #else
         rv = NS_InitEmbedding(greDir, nsnull);
+        #endif
         if (NS_FAILED(rv))
         {
             wxLogError(wxT("InitEmbedding Failed!"));
@@ -437,6 +446,11 @@
     if (!EnsureEmbedding())
         return FALSE;
 
+	nsCOMPtr<nsIComponentRegistrar> compReg;
+	nsresult rv2 = NS_GetComponentRegistrar(getter_AddRefs(compReg));
+	NS_ENSURE_SUCCESS(rv2, rv2);
+
+
 	nsCOMPtr <nsIEventQueueService> eventQService = do_GetService(kEventQueueServiceCID);
 	if(!eventQService)
 	{
@@ -458,11 +472,11 @@
     rv = NS_NewPromptServiceFactory(getter_AddRefs(promptFactory));
     if (NS_FAILED(rv)) return FALSE;
 	//use our own (native) prompts instead of the Mozilla XUL default ones
-	rv = nsComponentManager::RegisterFactory(kPromptServiceCID,
+	rv = compReg->RegisterFactory(kPromptServiceCID,
                                            "Prompt Service",
                                            "@mozilla.org/embedcomp/prompt-service;1",
-                                           promptFactory,
-                                           PR_TRUE); // replace existing
+                                           promptFactory /*,
+                                           PR_TRUE*/); // replace existing
 	if (NS_FAILED(rv)) {
 		return FALSE;
 	}
@@ -545,7 +559,11 @@
 	nsCOMPtr<nsPIDOMWindow> domWindowPrivate = do_QueryInterface(domWindow);
   // and the root window for that DOM window
 	nsCOMPtr<nsIDOMWindowInternal> rootWindow;
+#ifdef MOZILLA_VERSION_U
+	rootWindow = domWindowPrivate->GetPrivateRoot();
+#else
 	domWindowPrivate->GetPrivateRoot(getter_AddRefs(rootWindow));
+#endif
   
 	nsCOMPtr<nsPIDOMWindow> piWin(do_QueryInterface(rootWindow));
 
@@ -553,7 +571,11 @@
 		return FALSE;
 
 	nsCOMPtr<nsIChromeEventHandler> chromeHandler;
+#ifdef MOZILLA_VERSION_U 
+	chromeHandler =  piWin->GetChromeEventHandler();
+#else
 	piWin->GetChromeEventHandler(getter_AddRefs(chromeHandler));
+#endif
 
 	m_Mozilla->mEventReceiver = do_QueryInterface(chromeHandler);
 
diff -uNr wxmozilla-0.5.6/src/wxMozillaProtocolHandler.cpp wxmozilla-0.5.6.new/src/wxMozillaProtocolHandler.cpp
--- wxmozilla-0.5.6/src/wxMozillaProtocolHandler.cpp	2004-08-15 18:56:31.000000000 +0200
+++ wxmozilla-0.5.6.new/src/wxMozillaProtocolHandler.cpp	2006-12-10 22:53:50.000000000 +0100
@@ -20,7 +20,6 @@
 #include "wxMozillaComponents.h"
 
 //Mozilla
-#include "nsIServiceManagerUtils.h"
 #include "nsIEventQueue.h"
 #include "nsIEventQueueService.h"
 #include "netCore.h"
diff -uNr src/wxMozillaProtocolHandler.h src.new/wxMozillaProtocolHandler.h
--- wxmozilla-0.5.6/src/wxMozillaProtocolHandler.h	2004-04-22 11:08:22.000000000 +0200
+++ wxmozilla-0.5.6.new/src/wxMozillaProtocolHandler.h	2006-12-10 23:15:58.000000000 +0100
@@ -42,7 +42,6 @@
 #include "wx/filesys.h"
 
 //wxMozilla
-#include "helpers.h"
 #include "wxMozillaComponents.h"
 
 /**
